bowling.js

```diff
@@ -3,29 +3,128 @@ class Bowling {
     this.frames = [];
     this.currentframe = [];
     this.score = 0;
-    this.spare = false;
-    this.strike = false;
+    this.spare = "off";
+    this.sparebonus = "off";
+    this.strike = "off";
+    this.strikebonus = "off";
+    this.strikes = "off";
```

Comment:
You can use parallel assignment `[this.spare, this.sparebonus, this.strike, this.strikebonus, this.strikes] = ["off"]`